lib/journaled/enqueue.rb

```diff
@@ -0,0 +1,20 @@
+module Journaled
+  class << self
+    def enqueue!(*args)
+      on_enqueue.call(*args)
+    end
+
+    def on_enqueue(&block)
```

Comment:
If there's any unease about customizing the enqueue function itself, I can make this private and remove it from the README. The main argument I'd make for that is that it's actually dangerous to _not_ use DelayedJob, since Journaled jobs should be persisted in the same transaction as the actual persistence operation taking place. But, as a counter-argument, I'd point out that there are other DB-backed queues like [que](https://github.com/que-rb/que) that provide the same benefit, and regardless we may want to move into a world where this is backed by ActiveJob and can lean directly on the default ActiveJob config. In that world, maybe we just have big ⚠️ signs in the README to talk about how it's dangerous to not co-locate background queues with your business data.

Reply:
I'd say that until we have vetted alternatives that we can vouch provide the desired guarantees, we should probably lean away from making this flexibility part of the public API, yeah. And even then, it's probably a matter of us providing specific adapters for blessed job systems rather than a generalized roll-your-own facility.

Reply:
Makes sense! I'll go ahead and make that adjustment.

(For context, initially when I was exploring this change, I was hoping to write more of a callback that would allow us to adjust DJ's enqueue behavior at runtime. Cracking open this extension point was mostly a byproduct/afterthought.)

Reply:
Done!